blockstack_client/data.py

```diff
@@ -1225,73 +1272,90 @@ def data_put( blockstack_url, data, proxy=None, wallet_keys=None, data_privkey=N
+        if not parts['app']:
+            # profile data
+            ret = put_mutable(
+                parts['blockchain_id'], parts['data_id'], data,
+                proxy=proxy, wallet_keys=wallet_keys, **kw
+            )
+        else:
             if data_privkey is None:
-                raise ValueError("data_privkey is None")
+                raise ValueError('data_privkey is None')
 
             # app-specific data
             fields = parts['fields']
             version = fields.get('version', None)
-            ret = put_app_data( parts['blockchain_id'], fields['service_id'], fields['account_id'], parts['data_id'], data, data_privkey, proxy=proxy, version=version )
-
-        else:
-            # profile data
-            ret = put_mutable( parts['blockchain_id'], parts['data_id'], data, proxy=proxy, wallet_keys=wallet_keys, **kw ) 
+            ret = put_app_data(
+                parts['blockchain_id'], fields['service_id'], fields['account_id'],
+                parts['data_id'], data, data_privkey, proxy=proxy, version=version
+            )
 
         end = time.time()
 
-    if os.environ.get("BLOCKSTACK_TEST") == "1":
-        log.debug("[BENCHMARK] data_put %s" % (end - begin))
+    if BLOCKSTACK_TEST is not None:
+        log.debug('[BENCHMARK] data_put {}'.format(end - begin))
 
     return ret
 
 
-def data_delete( blockstack_url, proxy=None, wallet_keys=None, **kw ):
+def data_delete(blockstack_url, proxy=None, wallet_keys=None, **kw):
     """
     Delete data from a blockstack URL (be it mutable or immutable).
     """
-    parts = storage.blockstack_data_url_parse( blockstack_url )
-    assert parts is not None, "invalid url '%s'" % blockstack_url
+    parts = storage.blockstack_data_url_parse(blockstack_url)
+    assert parts is not None, 'invalid url "{}"'.format(blockstack_url)
 
     if parts['type'] == 'immutable':
-        return delete_immutable( parts['blockchain_id'], parts['fields']['data_hash'], data_id=parts['data_id'], proxy=proxy, wallet_keys=wallet_keys, **kw )
-    else:
-
-        ret = None
-        if parts['app']:
-            # app-specific delete
-            fields = parts['fields']
-            ret = delete_app_data(parts['blockchain_id'], fields['service_id'], fields['account_id'], parts['data_id'], proxy=proxy )
+        return delete_immutable(
+            parts['blockchain_id'], parts['fields']['data_hash'],
+            data_id=parts['data_id'], proxy=proxy, wallet_keys=wallet_keys, **kw
+        )
+
+    ret = None
+    if not parts['app']:
+        ret = delete_mutable(
+            parts['blockchain_id'], parts['data_id'],
+            proxy=proxy, wallet_keys=wallet_keys
+        )
+        return ret
 
-        else:
-            ret = delete_mutable( parts['blockchain_id'], parts['data_id'], proxy=proxy, wallet_keys=wallet_keys )
+    # app-specific delete
+    fields = parts['fields']
+    # BUG: required parameter "data_privkey" is not provided as argument
```

Comment:
Great catch!  I double-checked and it looks like `data_get`, `data_delete`, and `data_list` aren't covered by the integration test framework.  I'll make a note to cover them.